perf/istio-install/create_cluster.sh

```diff
@@ -136,12 +142,95 @@ gc beta container \
   --no-enable-basic-auth --cluster-version "${GKE_VERSION}" \
-  --machine-type "${MACHINE_TYPE}" --image-type "COS" --disk-type "pd-standard" --disk-size "${DISK_SIZE}" \
+  --issue-client-certificate \
+  --machine-type "${MACHINE_TYPE}" --image-type ${IMAGE} --disk-type "pd-standard" --disk-size "${DISK_SIZE}" \
   --scopes "${SCOPES}" \
   --num-nodes "${MIN_NODES}" --enable-autoscaling --min-nodes "${MIN_NODES}" --max-nodes "${MAX_NODES}" --max-pods-per-node "${MAXPODS_PER_NODE}" \
   --enable-stackdriver-kubernetes \
   --enable-ip-alias \
+  --metadata disable-legacy-endpoints=true \
   ${NETWORK_SUBNET} \
   --default-max-pods-per-node "${MAXPODS_PER_NODE}" \
   --addons "${ADDONS}" \
-  --enable-network-policy --enable-autoupgrade --enable-autorepair --labels csm=1,test-date=$(date +%Y-%m-%d),version=${ISTIO_VERSION},operator=user_${USER}
+  --enable-network-policy \
+  --workload-metadata-from-node=EXPOSED \
+  --enable-autoupgrade --enable-autorepair \
+  --labels csm=1,test-date=$(date +%Y-%m-%d),version=${ISTIO_VERSION},operator=user_${USER}
+
+NETWORK_NAME=$(basename "$(gcloud container clusters describe $CLUSTER --project $PROJECT_ID --zone=$ZONE \
+    --format='value(networkConfig.network)')")
+SUBNETWORK_NAME=$(basename "$(gcloud container clusters describe $CLUSTER --project $PROJECT_ID \
+    --zone=$ZONE --format='value(networkConfig.subnetwork)')")
+
+# Getting network tags is painful. Get the instance groups, map to an instance,
+# and get the node tag from it (they should be the same across all nodes -- we don't
+# know how to handle it, otherwise).
+INSTANCE_GROUP=$(gcloud container clusters describe $CLUSTER --project $PROJECT_ID --zone=$ZONE --format='flattened(nodePools[].instanceGroupUrls[].scope().segment())' |  cut -d ':' -f2 | tr -d "[:space:]")
+INSTANCE=$(gcloud compute instance-groups list-instances $INSTANCE_GROUP --project $PROJECT_ID \
+    --zone=$ZONE --format="value(instance)" --limit 1)
+NETWORK_TAGS=$(gcloud compute instances describe $INSTANCE --zone=$ZONE --project $PROJECT_ID --format="value(tags.items)")
+
+
+cat <<EOF > ${CLUSTER}/configmap-neg.yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: gce-config
+  namespace: kube-system
+data:
+  gce.conf: |
+    [global]
+    token-url = nil
+    # Your cluster's project
+    project-id = ${PROJECT_ID}
+    # Your cluster's network
+    network-name =  ${NETWORK_NAME}
+    # Your cluster's subnetwork
+    subnetwork-name = ${SUBNETWORK_NAME}
+    # Prefix for your cluster's IG
+    node-instance-prefix = gke-${CLUSTER}
+    # Network tags for your cluster's IG
+    node-tags = ${NETWORK_TAGS}
+    # Zone the cluster lives in
+    local-zone = ${ZONE}
+EOF
+
+
+cat <<EOF > ${CLUSTER}/configmap-galley.yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: istiod-asm
+  namespace: istio-system
+data:
+  galley.json: |
+      {
+      "EnableServiceDiscovery": true,
+      "SinkAddress": "meshconfig.googleapis.com:443",
```

Comment:
Can you make this endpoint configurable?

Reply:
Separate PR, it's been stuck for too long.